library/gen_837i.inc.php

```diff
@@ -0,0 +1,1946 @@
+* 
+* LICENSE: This Source Code is subject to the terms of the Mozilla Public License, v. 2.0.
+* See the Mozilla Public License for more details.
+* If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*
+* @package LibreHealth EHR
+* @author Terry Hill <teryhill@librehealth.io>
+* @link http://librehealth.io
+*
+* Please help the overall project by sending changes you make to the authors and to the LibreHealth EHR community.
+*
+*/
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+
+require_once("Claim.class.php");
+
+
+function gen_837I($pid, $encounter, &$log, $encounter_claim=false) {
+
+  $today = time();
+  $out = '';
+  $claim = new Claim($pid, $encounter);
+  $edicount = 0;
+
+  // This is true for the 5010 standard, false for 4010.
+  // x12gsversionstring() should be "005010X222A1" or "004010X098A1".
+  $CMS_5010 = strpos($claim->x12gsversionstring(), '5010') !== false;
+
+  $log .= "Generating claim $pid-$encounter for " .
+    $claim->patientFirstName()  . ' ' .
+    $claim->patientMiddleName() . ' ' .
+    $claim->patientLastName()   . ' on ' .
+    date('Y-m-d H:i', $today) . ".\n";
+
+
+
+  $out .= "ISA" .
+    "*" . $claim->x12gsisa01() .
+    "*" . $claim->x12gsisa02() .
+    "*" . $claim->x12gsisa03() .
+    "*" . $claim->x12gsisa04() .
+    "*" . $claim->x12gsisa05() .
+    "*" . $claim->x12gssenderid() .
+    "*" . $claim->x12gsisa07() .
+    "*" . $claim->x12gsreceiverid() .
+	// date of transmission "*030911" .
+	"*" . date('Ymd', $today) .
+    //Time of transmission "*1630" .
+	"*" . date('Hi', $today) .
+	"*" . ($CMS_5010 ? "^"     : "U"    ) .
+    "*" . ($CMS_5010 ? "00501" : "00401") .
+    "*000000001" .
+    "*" . $claim->x12gsisa14() .
+    "*" . $claim->x12gsisa15() .
+    "*:" .
+    "~\n";
+
+  $out .= "GS" .
+    "*HC" .
+    "*" . $claim->x12gsgs02() .
+    "*" . trim($claim->x12gs03()) .
+    "*" . date('Ymd', $today) .
+    "*" . date('Hi', $today) .
+    "*1" .
+    "*X" .
+   // "*" . $claim->x12gsversionstring() . 
+    "*" . "005010X223A2".
+    "~\n";
+
+  ++$edicount;
+  $out .= "ST" .
+    "*837" .
+    "*0021" .
+    //($CMS_5010 ? ("*" . $claim->x12gsversionstring()) : "") .
+	"*" . "005010X223A2".
+    "~\n";
+
+  ++$edicount;
+  $out .= "BHT" .
+    "*0019" .                             // 0019 is required here
+    "*00" .                               // 00 = original transmission
+    "*0123" .                             // reference identification
+    "*" . date('Ymd', $today) .           // transaction creation date
+    "*" . date('Hi', $today) .            // transaction creation time
+    ($encounter_claim ? "*RP" : "*CH") .  // RP = reporting, CH = chargeable
+    "~\n";
+
+  if (!$CMS_5010) {
```

Comment:
It was done in 2014 and the gen837 was the patern. It is being submitted and paid in it's current form but could stand going thru.